### DISCUSSION TITLE
Fixed Interest

### DISCUSSION THREAD

**Discussion by manju-rn** [ghostfolio/ghostfolio#3603] [Ideas]
The current Interest activity is not useful. A typical Fixed Interest would be
1. Initial Amount - say X
2. Date Invested - say mm/dd/yyy
3. Interest rate - say y%  per annum
4. Duration -  The duration of which this amount will mature, in days (months/years)
     OR Maturity date - must always be after Data Invested

The Tracking should  be on the daily basis where the amount is updated based on the time elapsed. 

_The Early withdrawal process where there is generally a fee involved could be done later_

**Comment by berkes** [comment]
In my case the actual interest is paid monthly at somewhat varying dates. The variation comes from shifts in weekends and holidays (first working day after ....). 

So, in my case I would need a way to *also* edit each automatically added interest activity, so that I can change dates if needed.

I can imagine another use-case where one wants to edit the amount posterior, because that can change due to calculation and/or rounding differences. These edits often merely change cents, though.

And another use-case where one may need to change interest rate every N time. So such an automation cannot rely on a single interest-rate, as often these change over time. And they then obviously don't affect past entries.

I'd think such a feature would be best part of a bigger "recurring events automation" setup. Where one can have an automation pre-fill (draft of some sort?) activities based on rules and types, and then review and edit and approve these entries. Interest would be the prime candidate for this, for me. As it's the most commonly recurring one.

**Comment by zypA13510** [comment]
### Workaround
1. Create a new manual asset profile for each term deposit.
    In the asset profile, expand **Historical Market Data** and you can manually "import" market data consisting of two points
    - 1 on the date invested
    - 1+x% on the maturity date
    Ghostfolio will generate a straight line across these two points.
2. Under account activities, create two orders of the manual asset:
    - Buy on the date invested, with quantity being the amount invested, and price 1.
    - Sell on the maturity date, with the same quantity and price 1+x%.

**Reply by MD0005** [reply]
Hopefully some day this can be created with simpler method in UI. Thank you for the idea!
